sunpy/map/mapbase.py

```diff
@@ -1327,7 +1314,7 @@ def _rotation_matrix_from_crota(self, crota_key='CROTA2'):
         If the specified key isn't present in the metadata, a default rotation
         of 0deg is returned.
         """
-        lam = self.scale[0] / self.scale[1]
+        lam = self.scale[1] / self.scale[0]
```

Comment:
```suggestion
        # lam = lat / lon
        lam = self.scale[1] / self.scale[0]
```

Reply:
This isn't always lat / lon, because whether axis 1/2 is lat/lon depends on the values of CTYPE{1,2}.

Reply:
(Carrying on from my blocking review comment here).

I don't think that's actually True. I think the rotation being in CROTA2 is actually only valid if latitude is the second axis (i.e. CTYPE2) see this relevant snippet:

![2021-12-20-165325_grim](https://user-images.githubusercontent.com/1391051/146803430-a4f9dcb7-c7dd-43ea-be45-8b34af135eef.png)

from [(Calbretta & Grisen 2002) (Paper II)[https://www.aanda.org/articles/aa/pdf/2002/45/aah3860.pdf].

Reply:
If you do the swaps 1 > 2 and 2 > 1 in eq 186, you still get lambda = CDELT2 / CDELT1, so I think what I've done is fine. This can be checked with the following code, that with this PR has expected axes limits as explained in the comments:
```python
import numpy as np
import sunpy.map
import matplotlib.pyplot as plt

# Test non-zero crota and unequal CDELT{1,2}
n = 10
data = np.random.rand(n, n)
header = {
    'CRVAL1': 0,
    'CRVAL2': 0,
    'CRPIX1': 0.5,
    'CRPIX2': 0.5,
    'NAXIS1': n,
    'NAXIS2': n,
    'CUNIT1': 'arcsec',
    'CUNIT2': 'arcsec',
    'CTYPE1': 'HPLT-TAN',  # lat
    'CTYPE2': 'HPLN-TAN',  # lon
    'CDELT1': 1,  # total axis length = 10 (lat)
    'CDELT2': 2,  # total axis length = 20 (lon)
    'CROTA2': 90,
    # Rotating by 90 degrees makes the transformation lon > lat, lat > -lon
    # So new axis limits should be:
    # lon: 0 - 10
    # lat: 0 - -20
    'DATE-OBS': '2020-01-01 00:00:00'
}
sunpy.map.Map(data, header).plot()
ax = plt.gca()
# Swap axis labels
if header['CROTA2'] == 90:
    c0 = ax.coords[0]
    c1 = ax.coords[1]
    c0.set_ticks_position('l')
    c0.set_ticklabel_position('l')
    c0.set_axislabel_position('l')
    c1.set_ticks_position('b')
    c1.set_ticklabel_position('b')
    c1.set_axislabel_position('b')

plt.show()
```